### DISCUSSION TITLE
Mechanics of application deletion without deleting k8s resources

### DISCUSSION THREAD

**Discussion by igorpalant** [argoproj/argo-cd#8125] [Q&A]
I have read documentation about resources-finalizer, but still a little confuse about mechanics of it. I am on version 1.8.6. My goal is to protect some of our services that are provisioned using Argo from removal if someone accidently removes application from Argo UI. 
 
https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/application.yaml
 #Add a this finalizer ONLY if you want these to cascade delete.
  finalizers:
    - resources-finalizer.argocd.argoproj.io

Our current application specs do not have this entry and on application deletion in UI, k8s resources are deleted. I would like to avoid it, but not sure I understand underlying mechanics. Thanks! 

Another question that I have about moving ArgoCD to another k8s cluster. I have Argo deployment that manages ~100 applications in various clusters. If I want to move ArgoCD to a different k8s cluster without disturbing deployed application - what should I do. Again, it comes from me missing understanding in underlying mechanics. Thanks in advance!

**Comment by jgwest** [comment]
Hi @igorpalant, I believe there is no way to prevent a user that is using the UI from accidentally deleting the Application child resources, with [the exception of preventing them from deleting using RBAC](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/).

RE: moving Argo CD to another cluster, I don't think we have documented an official way of doing thing, but my expectation is that the second Argo CD should automatically adopt the resources of the first Argo CD on the target k8s cluster, presuming that both Argo CD instances are setup in the same way.

**Comment by LinTechSo** [comment]
Hi. any updates?

**Reply by paulliss** [reply]
"[To preserve deployed resources](
https://argocd-applicationset.readthedocs.io/en/stable/Application-Deletion/), set .syncPolicy.preserveResourcesOnDeletion to true in the ApplicationSet." - Try it, works for me.

**Reply by lpsm-dev** [reply]
This only works if you use the ApplicationSet.

**Comment by kostis-codefresh** [comment]
>  If I want to move ArgoCD to a different k8s cluster without disturbing deployed application - what should I do

We have written a full guide here https://codefresh.io/blog/argocd-application-deletion-finalizers/

Check "Use case 4 – Moving or renaming an Argo CD application" section in the article. Although i think you will find the whole content interesting.
